core/access/src/main/java/uk/gov/gchq/gaffer/access/predicate/AccessPredicate.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.access.predicate;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import uk.gov.gchq.gaffer.user.User;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.function.BiPredicate;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "class")
+public abstract class AccessPredicate implements BiPredicate<User, String>, Serializable {
```

Comment:
Instead of making this a BiPredicate, could we make it a Predicate<User> which would allow use to make use of the existing Koryphe Functions and Predicates?

The second adminUser argument could be a class field of the predicate as it's unlikely to change.